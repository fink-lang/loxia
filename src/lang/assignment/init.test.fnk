{fink2js} = import '../../testing/generate.fnk'
{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'


describe 'assignment', fn:
  it 'compiles', fn:
    expect
      fink2js '
        assign = 1
        # simple
        assign2 = assign

        no_clash_with_await = 13

        assign_lang_const1 = true
        assign_lang_const2 = false

        multi_line_assign = 123
          + 234 +
          -567
          - 1111
      '
      to_match_snapshot



describe 'list destructuring', fn:
  it 'compiles leading, last', fn:
    expect
      fink2js "
        [...leading, last] = '1234'
      "
      to_match_snapshot

  it 'compiles first, some, last', fn:
    expect
      fink2js "
        # enhanced destructuring
        [foo, bar, ...spam, ni, shrub] = '123'

        [a, , ..., b, c] = '123'
      "
      to_match_snapshot
