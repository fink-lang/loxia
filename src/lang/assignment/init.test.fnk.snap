// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assignment compiles 1`] = `
"export const assign = 1;
// simple
export const assign2 = assign;
export const no_clash_with_await = 13;
export const assign_lang_const1 = true;
export const assign_lang_const2 = false;
export const multi_line_assign = 123 + 234 + (-567 - 1111);"
`;

exports[`unpacking lists compiles first, some, last 1`] = `
"let _do_result;

{
  const [...ˆitems_1] = \`123\`;
  _do_result = [ˆitems_1, ˆitems_1.slice(2, -2), ˆitems_1.slice(-2)];
}
// enhanced destructuring
const [[foo, bar], spam, [ni, shrub]] = _do_result;
_do_result = undefined;

let _do_result2;

{
  const [...ˆitems_2] = \`123\`;
  _do_result2 = [ˆitems_2, ˆitems_2.slice(2, -2), ˆitems_2.slice(-2)];
}
const [[a,,],, [b, c]] = _do_result2;
_do_result2 = undefined;"
`;

exports[`unpacking lists compiles leading, last 1`] = `
"let _do_result;

{
  const [...ˆitems_1] = \`1234\`;
  _do_result = [ˆitems_1.slice(0, -1), ˆitems_1.slice(-1)];
}
const [leading, [last]] = _do_result;
_do_result = undefined;"
`;
